specification/dsdl/conventions.tex

```diff
@@ -0,0 +1,115 @@
+\section{Conventions and recommendations}
+
+This section is dedicated to conventions and recommendations
+intended to help data type designers maintain a consistent style across the ecosystem
+and avoid some common pitfalls.
+Some of the conventions are mandatory to follow, others are optional.
```

Comment:
I've added some comments below but generally I don't like mixing sparse bits of normative specification into larger blocks of non-normative. The former tends to get lost. Can we refactor this to have a short normative/mandatory section followed by the longer non-normative/optional sections?

Reply:
I think I have addressed this in https://github.com/UAVCAN/specification/pull/46/commits/dbdc89b9d1b66f7ce292898bdb8e72df5a68a3aa